### PR TITLE
ENH: Optimize replace to avoid copying when not necessary

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by lithomas1** [comment]
Does this fix any of the existing cases in ``test_replace``?

**Comment by phofl** [comment]
No, not yet

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -523,6 +525,8 @@ def replace(
         inplace: bool = False,
         # mask may be pre-computed if we're called from replace_list
         mask: npt.NDArray[np.bool_] | None = None,
+        original_blocks: list[Block] = [],
```
Comment (lithomas1) [inline pandas/core/internals/blocks.py:528]: Do we need to take a list here, since we're only setting the ref to 1 block?

Comment (phofl) [reply pandas/core/internals/blocks.py:528]: I want to use the same as in https://github.com/pandas-dev/pandas/pull/50428, but want to get this merged before

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -575,22 +581,36 @@ def replace(
...
             #  replace_list instead of replace.
-            return [self] if inplace else [self.copy()]
+            if using_cow:
+                return [self.copy(deep=False)]
+            else:
+                return [self] if inplace else [self.copy()]
 
         if mask is None:
             mask = missing.mask_missing(values, to_replace)
         if not mask.any():
```
Comment (lithomas1) [inline pandas/core/internals/blocks.py:591]: If we pass an axis(I think``axis=0`` should work, need to be careful of 1-D values/mask though), than we should have enough info to split the block.

Comment (phofl) [reply pandas/core/internals/blocks.py:591]: Yep agree, but right now this is not that high on my list of priorities. With the ref tracking how it is right now, this only defers the copy a little bit

**Review by lithomas1 — APPROVED** [review]
LGTM. Can you update the list on #49473?

**Comment by lithomas1** [comment]
thanks @phofl

**Pull request merged**